src/Microsoft.AspNet.Mvc.TagHelpers/Internal/TagHelperContextExtensions.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.Mvc.TagHelpers.Internal;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Logging;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
+{
+    /// <summary>
+    /// Utility related extensions for <see cref="TagHelperContext"/>.
+    /// </summary>
+    public static class TagHelperContextExtensions
+    {
+        /// <summary>
+        /// Determines whether a <see cref="ITagHelper" />'s required attributes are present, non null, non empty, and
+        /// non whitepsace.
+        /// </summary>
+        /// <param name="context">The <see cref="TagHelperContext"/>.</param>
+        /// <param name="requiredAttributes">The attributes the <see cref="ITagHelper" /> requires in order to run.</param>
+        /// <param name="logger">An optional <see cref="ILogger"/> to log warning details to.</param>
+        /// <returns>A <see cref="bool"/> indicating whether the <see cref="ITagHelper" /> should run.</returns> 
+        public static bool AllRequiredAttributesArePresent(
```

Comment:
That method already existed before this PR so I'm not inclined to change it now. It will in all likelihood disappear completely once the script tag helper gets globbing support anyway.